smurfs/src/components/Form.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { connect } from "react-redux";
+import {
+  inputChange,
+  submit,
+  updateSmurfs,
+  getSmurfs
+} from "../state/actionCreators";
+
+export function Form({ form, inputChange, submit, updateSmurfs, getSmurfs }) {
+  const handleChange = e => {
+    inputChange(e.target.name, e.target.value);
+  };
+  const handleSubmit = e => {
+    e.preventDefault();
+    submit();
+    const form = {
+      name: e.target.name.value,
+      age: e.target.age.value,
+      height: e.target.height.value
+    };
+    updateSmurfs(form);
+    getSmurfs();
+  };
+
+  return (
+    <form className="formContainer" onSubmit={handleSubmit}>
+      <p>Add a new Smurf</p>
+      <input
+        type="text"
+        name="name"
+        value={form.name}
+        placeholder="What shall we call your new smurf?"
+        onChange={handleChange}
+        className="formInput"
+      />
+      <input
+        type="text"
+        name="age"
```

Comment:
a Form component is a perfect use case for a component state, since other components do not need the form data